test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java

```diff
@@ -20,60 +20,106 @@
 package org.elasticsearch.test;
 
 import org.elasticsearch.Version;
+import org.elasticsearch.common.collect.Tuple;
 
 import java.lang.reflect.Field;
 import java.lang.reflect.Modifier;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Random;
-import java.util.Set;
-import java.util.stream.Collectors;
+import java.util.regex.Pattern;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.unmodifiableList;
 
 /** Utilities for selecting versions in tests */
 public class VersionUtils {
+    /**
+     * Matches names of versions that we process. Intentionally skips things like CURRENT.
```

Comment:
Now that I think about it, maybe I should just skip `CURRENT` and drop the regex entirely.